syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -450,3 +451,72 @@ def _restore_from_state(self, state: dict):
 
     def set_random_state(self, random_state: np.random.RandomState):
         self.random_state = random_state
+
+
+class FilterDuplicatesMixin:
```

Comment:
the mixin should not have functions overlapping with BaseSearcher

Reply:
Yes, so this would basically be multiple inheritance then.

The problem with moving this into BaseSearcher:
- BaseSearcher is really supposed to be just an API. David would rightly reject putting code in there
- Some of the searchers do this `allow_duplicates` in a different way

Is multiple inheritance really so evil?

Reply:
OK, I think what I can do, is I can integrate this code into `SearcherWithRandomSeed`. This way, the `BaseSearcher` remains clean.

Reply:
Or better: make it a subclass of that one, this is best